python/src/apx_agent/_eval.py

```
@@ -243,3 +251,221 @@ def _predict(inputs: Any) -> str:
+    }
+
+    def predict(inputs: Any) -> str:
+        import urllib.request
+
+        question = _extract_question(inputs)
+        body = _json.dumps({"input": question, "stream": stream}).encode()
+        req = urllib.request.Request(url, data=body, headers=headers, method="POST")
+
+        if stream:
+            full_text = ""
+            with urllib.request.urlopen(req, timeout=timeout) as resp:
```

Ruff (0.15.13)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)